docs/api-reference/next/router.md

```diff
@@ -112,6 +112,8 @@ export default function Page() {
 }
 ```
 
+Note: when navigating to the same page in Next.js the page's state will not be reset by default as the top-level react component is the same. You can manually ensure the state is updated using `useEffect` or adding a `key` prop with the `asPath` to the components who's state should be reset.
```

Comment:
```suggestion
> **Note:** When navigating to the same page in Next.js, the page's state **will not** be reset by default, as the top-level React component is the same. You can manually ensure the state is updated using `useEffect` or by adding a `key` prop with the `asPath` to the components who's state should be reset.
```

This part "adding a `key` prop with the `asPath` to the components who's state should be reset." is still confusing to me, not sure how to reword.

Reply:
Maybe we shouldn't mention this since it also clears the DOM and is less optimal than using `useEffect` to update the state 🤔